# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: startupcache/StartupCache.h
# Commit: db2b0b9a9886
# Full Hash: db2b0b9a98861da3fe3266e08dd53b2d5a64dfe7
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-02-12 09:32:01
# Regressor Bug: 1607505
# File Overlap Count: 3
# Description:
#   Bug 1607505 - Start writing StartupCache earlier in shutdown r=froydnj
#   
#   The initial thought for getting the StartupCache out of the shutdown
#   path was to simply not write it during shutdown, as it should write
#   60 seconds after startup, and the theory was that if the user shut
# ==============================================================================

diff -r d0226991e792 -r db2b0b9a9886 startupcache/StartupCache.h
--- a/startupcache/StartupCache.h	Tue Feb 11 21:05:25 2020 +0000
+++ b/startupcache/StartupCache.h	Tue Feb 11 19:14:35 2020 +0000
@@ -163,6 +163,10 @@
   // Removes the cache file.
   void InvalidateCache(bool memoryOnly = false);
 
+  // For use during shutdown - this will write the startupcache's data
+  // to disk if the timer hasn't already gone off.
+  void MaybeInitShutdownWrite();
+
   // Signal that data should not be loaded from the cache file
   static void IgnoreDiskCache();
 
@@ -203,6 +207,7 @@
   void WaitOnWriteThread();
   void WaitOnPrefetchThread();
   void StartPrefetchMemoryThread();
+  void MaybeSpawnWriteThread();
 
   static nsresult InitSingleton();
   static void WriteTimeout(nsITimer* aTimer, void* aClosure);